trieste/models/keras/models.py

```diff
@@ -106,15 +107,28 @@ def __init__(
         :param continuous_optimisation: If True (default), the optimizer will keep track of the
             number of epochs across BO iterations and use this number as initial_epoch. This is
             essential to allow monitoring of model training across BO iterations.
+        :param compile_args: Keyword arguments to pass to the ``compile`` method of the
+            Keras model (:class:`~tf.keras.Model`).
+            See https://keras.io/api/models/model_training_apis/#compile-method for a
+            list of possible arguments. The ``optimizer``, ``loss`` and ``metrics`` arguments
+            must not be included.
         :raise ValueError: If ``model`` is not an instance of
-            :class:`~trieste.models.keras.KerasEnsemble` or ensemble has less than two base
-            learners (networks).
+            :class:`~trieste.models.keras.KerasEnsemble`, or ensemble has less than two base
+            learners (networks), or `compile_args` contains disallowed arguments.
         """
         if model.ensemble_size < 2:
             raise ValueError(f"Ensemble size must be greater than 1 but got {model.ensemble_size}.")
 
         super().__init__(optimizer)
 
+        if compile_args is None:
+            compile_args = {}
+
+        if "optimizer" in compile_args or "loss" in compile_args or "metrics" in compile_args:
```

Comment:
(following is shorter, but possibly less clear?)

```suggestion
        if not {"optimizer", "loss", "metrics"}.isdisjoint(compile_args):
```